src/picker_gui.h

```diff
@@ -0,0 +1,224 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file picker_gui.h Functions/types etc. related to the picker GUI. */
+
+#ifndef PICKER_GUI_H
+#define PICKER_GUI_H
+
+#include "querystring_gui.h"
+#include "sortlist_type.h"
+#include "stringfilter_type.h"
+#include "strings_type.h"
+#include "timer/timer.h"
+#include "timer/timer_game_calendar.h"
+#include "timer/timer_window.h"
+#include "window_gui.h"
+#include "window_type.h"
+
+struct PickerItem {
+	uint32_t grfid;
+	uint16_t local_id;
+	int class_index;
+	int index;
+
+	inline auto operator<=>(const PickerItem &other) const
+	{
+		if (auto cmp = this->grfid <=> other.grfid; cmp != 0) return cmp;
+		return this->local_id <=> other.local_id;
+	}
+};
+
+/** Class for PickerClassWindow to collect information and retain state. */
+class PickerCallbacks {
+public:
+	explicit PickerCallbacks(const std::string &ini_group);
+	virtual ~PickerCallbacks();
+
+	virtual void Close([[maybe_unused]] int data) { }
```

Comment:
I do not think it makes much sense annotating that the parameters are unused in here. Or is it required that both the declaration and definition are in sync? If so, then I've missed a lot. If not, then I've also not added a comment to each of the locations where it should be removed.
```suggestion
	virtual void Close(int data) { }
```